merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/services/GetSimulationResultsAction.java

```diff
@@ -113,9 +122,11 @@ public Map<String, List<Violation>> getViolations(final PlanId planId, final Sim
           activity.arguments(),
           Window.between(activityOffset, activityOffset.plus(activity.duration()))));
     }
-
-    final var discreteProfiles = new HashMap<String, DiscreteProfile>(results.discreteProfiles.size());
-    for (final var entry : results.discreteProfiles.entrySet()) {
+    final var _discreteProfiles = results$
```

Comment:
Can you justify this change?

Reply:
Essentially, it was necessary to remove `getViolations()`'s dependence on the `SimulationResults` argument, since we want to call this method without having run the simulation, so that we can return an empty violation map if the sim was uncached but `CachedSimulationService` was being used. Instead, we are getting an `Optional<SimulationResults` object from the new getter in the `SimulationService` interface.

Reply:
One thought I have is that in maintaining backwards compatibility, we now load simulation results twice (in the case where `getViolations` is called from `GetSimulationResultsAction::run`). If we weren't planning on getting rid of that call, I would make a fuss about it, but as it stands I'm okay with leaving that in here until we remove that call in a follow-up PR :+1:

Reply:
That is a good point. The backwards compatibility was a bit of an afterthought. I suppose the double loading is somewhat inherent to allowing these queries to be made separately though.

Reply:
One way to "shim in" the backwards compatibility, without loading the data twice, would be to have two `getViolations`methods, one with one parameter `getViolations(PlanId)`, and the other with two parameters `getViolations(PlanId, SimulationResults)`. The first would load the results and call the second, whereas the "old" workflow would continue more or less unchanged.

If we wanted to keep both, I would push for this approach, but since we're removing one, this would add some unnecessary clutter.

Reply:
I agree it would add some unnecessary clutter, especially since the removal of the deprecated API *should* happen within the same release (ie 0.12.1). I'll leave it as is then, barring additional feedback!